debugging/netcoreapp/dockerDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,242 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as path from 'path';
+import { CancellationToken, DebugConfiguration, DebugConfigurationProvider, ProviderResult, WorkspaceFolder } from 'vscode';
+import { DockerManager, LaunchResult } from './dockerManager';
+import { FileSystemProvider } from './fsProvider';
+import { NetCoreProjectProvider } from './netCoreProjectProvider';
+import { OSProvider, PlatformType  } from './osProvider';
+import { Prerequisite } from './prereqManager';
+
+interface DockerDebugBuildOptions {
+    context?: string;
+    dockerfile?: string;
+    tag?: string;
+    target?: string;
+}
+
+interface DockerDebugRunOptions {
+    containerName?: string;
+    os?: PlatformType;
+}
+
+interface DockerDebugOptions {
+    appFolder?: string;
+    appOutput?: string;
+    appProject?: string;
+    build?: DockerDebugBuildOptions;
+    run?: DockerDebugRunOptions;
+}
+
+interface DebugConfigurationBrowserBaseOptions {
+    enabled?: boolean;
+    command?: string;
+    args?: string;
+}
+
+interface DebugConfigurationBrowserOptions extends DebugConfigurationBrowserBaseOptions {
+    windows?: DebugConfigurationBrowserBaseOptions;
+    osx?: DebugConfigurationBrowserBaseOptions;
+    linux?: DebugConfigurationBrowserBaseOptions;
+}
+
+interface DockerDebugConfiguration extends DebugConfiguration {
+    dockerOptions?: DockerDebugOptions;
+}
+
+export class DockerDebugConfigurationProvider implements DebugConfigurationProvider {
+    constructor(
+        private readonly dockerManager: DockerManager,
+        private readonly fsProvider: FileSystemProvider,
+        private readonly osProvider: OSProvider,
+        private readonly netCoreProjectProvider: NetCoreProjectProvider,
+        private readonly prerequisite: Prerequisite) {
+    }
+
+    public provideDebugConfigurations(folder: WorkspaceFolder | undefined, token?: CancellationToken): ProviderResult<DebugConfiguration[]> {
+        return [
+            {
+                name: 'Docker: Launch .NET Core',
+                type: 'docker-netcoreapp',
```

Comment:
I think I originally chose `netcoreapp` because it's the name of the target platform for .NET Core (e.g. in the project file).  I agree that `docker-coreclr` is more consistent with the naming of the debug adaptor for .NET Core in VS Code.  Changed.